### PR TITLE
get server metrics data by id

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 1]
```diff
@@ -342,40 +332,11 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
-         }
-      }
-      return generateMetricsData(valueunits);
+      return assetService.getPduMetricsDataById(assetID);
    }
 
    //starttime miliseconds.
-   @ResponseStatus(HttpStatus.OK)
-   @RequestMapping(value = "/{id}/serversensordata", method = RequestMethod.GET)
+   @RequestMapping(value = "server/{id}/realtimedata", method = RequestMethod.GET)
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: host/

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+      Map<String, Map<String, String>> pduMetrics = metricFormula.get(FlowgateConstant.PDU);
+      if(pduMetrics != null && !pduMetrics.isEmpty()) {
+         List<String> metricNames = new ArrayList<String>();
+         metricNames.add(MetricName.PDU_TOTAL_POWER);
+         metricNames.add(MetricName.PDU_TOTAL_CURRENT);
+         metricNames.add(MetricName.PDU_APPARENT_POWER);
+         metricNames.add(MetricName.PDU_CURRENT);
+         metricNames.add(MetricName.PDU_VOLTAGE);
+         int index = 1;
+         for(String pduId : pduMetrics.keySet()) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Let's use pduid as the index.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+            String metricName = sensorFormula.getKey();
+            List<ValueUnit> valueUnits = generateSensorValueUnit(assetIdAndRealtimeDataMap,
+                  starttime, locationAndIdMap, metricName);
+            result.addAll(generateServerSensorMetricData(valueUnits, metricName));
+         }
+      }
+      return result;
+   }
+
+   private List<MetricData> generateServerPduMetricData(List<ValueUnit> valueUnits, int index){
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: change to assetId

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 3 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: done

**Pull request merged**